components/formats-bsd/src/loci/formats/in/OMETiffReader.java

```diff
@@ -1056,7 +1063,14 @@ public MetadataStore getMetadataStoreForConversion() {
     int realSeries = getSeries();
     for (int i=0; i<getSeriesCount(); i++) {
       setSeries(i);
-      store.setPixelsBinDataBigEndian(new Boolean(!isLittleEndian()), i, 0);
+      if (meta.getPixelsBinDataCount(i) > 0) {
+        for (int j=0; j<meta.getPixelsBinDataCount(i); j++) {
+          store.setPixelsBinDataBigEndian(new Boolean(!isLittleEndian()), i, j);
+        }
+      }
+      else {
+        store.setPixelsBigEndian(Boolean.TRUE, i);
```

Comment:
I don't think this should be in an else block--I think it should be set unconditionally irrespective of whether any BinData elements are present.  And I think it should be set using `!isLittleEndian()` for consistency and to match what's set in core metadata.